sql/tables.go

```diff
@@ -342,6 +342,13 @@ type AutoIncrementTable interface {
 	AutoIncrementSetter(*Context) AutoIncrementSetter
 }
 
+// AutoIncrementGetter provides support for reading a table's AUTO_INCREMENT value.
+// This can include tables that don't implement AutoIncrementTable if the table is a read-only snapshot
+// of an auto-incremented table.
+type AutoIncrementGetter interface {
```

Comment:
Please embed this in the `AutoIncrementTable` interface to make the relationship clearer